src/libstd/sync/barrier.rs

```diff
@@ -141,7 +141,7 @@ impl Barrier {
     pub fn wait(&self) -> BarrierWaitResult {
         let mut lock = self.lock.lock().unwrap();
         let local_gen = lock.generation_id;
-        lock.count += 1;
+        lock.count.wrapping_add(1);
```

Comment:
You need `lock.count = lock.count.wrapping_add(1)` (`wrapping_add` returns the new value instead of modifying in-place).